### PR TITLE
Item stats#58

### LINKED ISSUE #58: User Story 15, Items Index Page Statistics
```
As any kind of user on the system
When I visit the items index page ("/items")
I see an area with statistics:
- the top 5 most popular items
- the bottom 5 least popular items
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #60

depends on #59

### PR DESCRIPTION
**Pull request by Bradniedt**
-Adds feature test for all types of users seeing the item statistics
-Adds model test for five_popular method. (this was originally two methods - top_five_popular and bottom_five_popular, but the only difference was their order ASC/DESC, so I refactored them into one method)
-Implements functionality in the item model and item index view.
-All tests passing except for the enable toggle that @saleratus said is taken care of elsewhere.
-99.76% coverage

closes #58

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -3,8 +3,12 @@ class ItemsController < ApplicationController
   def index
    if current_merchant?
      @items = current_user.items
+     @top_five_items = Item.five_popular('desc')
```
Comment (asmolentzov) [inline app/controllers/items_controller.rb]: You should be able to DRY this up by moving @top_five_items and @bottom_five_items below the if/else/end. That way it'll be available regardless of user type.

File: spec/features/item_statistics_spec.rb [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    @item_2 = create(:item)
+    @item_3 = create(:item)
+    @item_4 = create(:item)
+    @item_5 = create(:item)
+    @item_6 = create(:item)
+    @item_7 = create(:item)
+    @order_item_1 = create(:fulfilled_order_item, item: @item_1) #add a times loop to build these?
+    @order_item_2 = create(:fulfilled_order_item, item: @item_1)
+    @order_item_3 = create(:fulfilled_order_item, item: @item_1)
+    @order_item_3 = create(:fulfilled_order_item, item: @item_1)
```
Comment (asmolentzov) [inline spec/features/item_statistics_spec.rb]: You've got the same variable name here on a bunch of these! Looks like the variable name doesn't really matter for this, so you could just go ahead and get rid of them entirely if you wanted.

File: spec/models/item_spec.rb [hunk 3]
```diff
@@ -18,22 +18,66 @@
...
+        item_2 = create(:item)
+        item_3 = create(:item)
+        item_4 = create(:item)
+        item_5 = create(:item)
+        item_6 = create(:item)
+        item_7 = create(:item)
+        order_item_1 = create(:fulfilled_order_item, item: item_1) #add a times loop to build these?
+        order_item_2 = create(:fulfilled_order_item, item: item_1)
+        order_item_3 = create(:fulfilled_order_item, item: item_1)
+        order_item_3 = create(:fulfilled_order_item, item: item_1)
```
Comment (asmolentzov) [inline spec/models/item_spec.rb]: Same thing with variable names here

File: spec/models/item_spec.rb [hunk 4]
```diff
@@ -18,22 +18,66 @@
...
+        order_item_3 = create(:fulfilled_order_item, item: item_2)
+        order_item_3 = create(:fulfilled_order_item, item: item_2)
+        order_item_3 = create(:fulfilled_order_item, item: item_4)
+        order_item_3 = create(:fulfilled_order_item, item: item_4)
+        order_item_3 = create(:fulfilled_order_item, item: item_6)
+        order_item_3 = create(:fulfilled_order_item, item: item_6)
+        order_item_3 = create(:fulfilled_order_item, item: item_5)
+        order_item_3 = create(:fulfilled_order_item, item: item_7)
+
+        expect(Item.five_popular('desc')).to include(item_1)
```
Comment (asmolentzov) [inline spec/models/item_spec.rb]: Would it make more sense to test for the exact array here? I guess the user story doesn't say that they have to display in order, but that's what we're doing anyways!

File: app/controllers/items_controller.rb (hunk 1 above)
Comment (Bradniedt) [reply app/controllers/items_controller.rb]: Nice! I'll add this!

File: spec/features/item_statistics_spec.rb (hunk 2 above)
Comment (Bradniedt) [reply spec/features/item_statistics_spec.rb]: Face palm. Thank you for catching this.

**Comment by danbriechle** [comment]
Spoke via slack. Looks good!
